hvplot/converter.py

```diff
@@ -1819,11 +1828,94 @@ def method_wrapper(ds, x, y):
+                    f"datashaded points; reverting to 'mouse'."
+                )
+
+            stream = PointerXY
+            if len(self.data) > 10000:
+                stream = Tap
+                param.main.param.warning(
+                    'Hovering over datashaded points is slow for large datasets; '
+                    'tap on the plot to see a hover tooltip over desired point.'
+                )
+
+            inspector = inspect_points.instance(
+                streams=[stream], transform=self._datashade_hover_transform
+            )
+            processed *= inspector(processed).opts(
+                size=10,
+                alpha=0,
+                tools=['hover'],
+                hover_mode=self.hover_mode,
+                hover_tooltips=self.hover_tooltips,
+                hover_formatters=self.hover_formatters,
+            )
+
         opts = filter_opts(eltype, dict(self._plot_opts, **style), backend='bokeh')
         layers = self._apply_layers(processed).opts(eltype, **opts, backend='bokeh')
         layers = _transfer_opts_cur_backend(layers)
         return layers
 
+    def _datashade_hover_transform(self, df):
+        if not len(df):
+            return df
+
+        # show at least the x and y columns
+        cols = self.hover_cols.copy()
+        if self.x not in cols:
+            cols.append(self.x)
+        if self.y not in cols:
+            cols.append(self.y)
+
+        # handle aggregator, e.g. ds.sum('column') or ds.count_cat('column')
+        agg_col = None
+        agg_series_map = {}
+        if self.aggregator and not isinstance(self.aggregator, str) and self.aggregator.column:
+            agg_col = self.aggregator.column
+            agg_op = type(self.aggregator).__name__
+            if hasattr(df, agg_op):  # df.sum/df.count
+                agg_value = df.agg({agg_col: agg_op})
+            elif agg_op == 'count_cat':
+                agg_value = df[agg_col].value_counts()
+
+            if agg_col in cols:
+                cols.remove(agg_col)
+        else:
+            key = 'Count'
+            for i in range(1, 10):
+                if key in df.columns:
+                    key = f'Count_{i}'
+                else:
+                    break
+            agg_value = pd.Series([len(df)], index=[key])
+
+        # take the mean of numeric columns
+        num_series = df[cols].select_dtypes(include=['number']).mean()
+        if len(num_series):
+            agg_series_map['number_cols'] = num_series
+
+        # take the first value of object columns
+        obj_series = df[cols].select_dtypes(exclude=['number']).iloc[0]
+        if len(obj_series):
+            agg_series_map['object_cols'] = obj_series
+
+        # to preserve order of other columns, add this last
+        agg_series_map[agg_col] = agg_value
+
+        # concat all series into a single dataframe which has one row
+        df_hover = pd.concat(agg_series_map.values()).to_frame().transpose()
```

Comment:
This especially, without the comment, would be really confusing of what all the methods are doing.

Reply:
Agree. But why is this needed? Why is a `pd.Series` not good enough?  We could add an extra line explaining this:
```suggestion
        # This is needed as the function used in `inspect_points.transform` must return a DataFrame.
        df_hover = pd.concat(agg_series_map.values()).to_frame().transpose()
```

At some point, a person could look at this line and remove `.to_frame().transpose()`, and if no tests fails people could think it was not a needed conversion.